delira/training/predictor.py

```diff
@@ -0,0 +1,277 @@
+import logging
+
+import numpy as np
+from tqdm import tqdm
+
+from ..data_loading import BaseDataManager
+
+logger = logging.getLogger(__name__)
+
+
+class Predictor(object):
+    """
+    Defines an API for Predictions from a Network
+
+    See Also
+    --------
+    :class:`PyTorchNetworkTrainer`
+
+    """
+
+    # static variable to prevent certain attributers from overwriting
+    __KEYS_TO_GUARD = []
+
+    def __init__(self, model, key_mapping: dict, 
+                 convert_batch_to_npy_fn=lambda x: x,
```

Comment:
lambda x: x no longer captures the desired default behaviour (which seems to be [discard *args and pass **kwargs) of returned model.call()](https://github.com/justusschock/delira/blob/trainer_refactoring/delira/training/predictor.py#L113). Either make it positional arg or change default setting. Also, the naming is no longer fitting (we convert the tensors inside the dict entries to np.ndarrays if required).